_posts/2020-12-09-#20477.md

```diff
@@ -10,13 +10,29 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
+* When all inbound connection slots are occupied and a new inbound connection is received, the node attempts to evict an existing inbound connection in order to open up a slot.
+* However, any eviction logic is susceptible to [eclipse attacks](https://www.avivz.net/pubs/15/sec15-paper-heilman_final.pdf). An attacker may try to make lots of new inbound connections to us, and if we always evict another honest node, we will eventually be eclipsed.
+* `AttemptToEvictConnection` in `src/net.cpp` contains logic to increase the cost of such attacks. Among other things, in order to not get their own previous connections evicted when they make new inbounds, the attacker would need to:
+	- Obtain IP addresses in various net groups
+	- Be amongst the fastest of our peers (geographic cost)
+	- Provide us novel transactions risking fees
+	- Provide us with novel blocks
```

Comment:
Possibly have a look at or link to net.h::L1039-1050 for more info / exact definitions of the requirements wrt txns (nLastTXTime) and blocks (nLastBlockTime)

Reply:
For info, these eviction criteria were behind the choice of some of the data columns displayed in  -netinfo (e.g. `watch ./src/bitcoin-cli -netinfo 4`)

Reply:
I think this can be left for the readers to figure out.